net/golioth/CMakeLists.txt

```
@@ -12,23 +12,20 @@ zephyr_library_sources_ifdef(CONFIG_GOLIOTH_SETTINGS settings.c)
 zephyr_library_sources_ifdef(CONFIG_GOLIOTH_SYSTEM_CLIENT system_client.c)
 
 if(CONFIG_GOLIOTH_AUTH_METHOD_CERT)
-  foreach(type IN ITEMS ca crt key)
-    string(TOUPPER ${type} type_upper)
-    set(path ${CONFIG_GOLIOTH_SYSTEM_CLIENT_${type_upper}_PATH})
+  set(path ${CONFIG_GOLIOTH_SYSTEM_CLIENT_CA_PATH})
 
-    if(NOT IS_ABSOLUTE ${path})
-      if(EXISTS ${APPLICATION_SOURCE_DIR}/${path})
-        set(path ${APPLICATION_SOURCE_DIR}/${path})
-      elseif(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${path})
-        set(path ${CMAKE_CURRENT_SOURCE_DIR}/${path})
-      else()
-        message(FATAL_ERROR "${path} configured by CONFIG_GOLIOTH_SYSTEM_CLIENT_${type_upper}_PATH does not exist")
-      endif()
+  if(NOT IS_ABSOLUTE ${path})
+    if(EXISTS ${APPLICATION_SOURCE_DIR}/${path})
+      set(path ${APPLICATION_SOURCE_DIR}/${path})
+    elseif(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${path})
+      set(path ${CMAKE_CURRENT_SOURCE_DIR}/${path})
+    else()
+      message(FATAL_ERROR "${path} configured by CONFIG_GOLIOTH_SYSTEM_CLIENT_CA_PATH does not exist")
     endif()
+  endif()
 
-    generate_inc_file_for_target(app ${path}
-      ${ZEPHYR_BINARY_DIR}/include/generated/golioth-systemclient-${type}.inc)
-  endforeach()
+  generate_inc_file_for_target(app ${path}
+    ${ZEPHYR_BINARY_DIR}/include/generated/golioth-systemclient-${type}.inc)
```

```suggestion
    ${ZEPHYR_BINARY_DIR}/include/generated/golioth-systemclient-ca.inc)
```

I don't think type is being set now that the `foreach` command is removed. The result is that the generated file is missing the `ca` in its title:

```
golioth-systemclient-.inc
golioth-systemclient-crt.inc
golioth-systemclient-key.inc
```

Which results in a build error:
```
/home/mike/golioth-zephyr-workspace/modules/lib/golioth/net/golioth/system_client.c:50:10: fatal error: golioth-systemclient-ca.inc: No such file or directory
   50 | #include "golioth-systemclient-ca.inc"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

---

Thanks! I had been trying to reproduce this locally with no luck. Turns out I had a stale cached version of `prj.conf` in my editor that was displaying `CONFIG_GOLIOTH_AUTH_METHOD_CERT=y` without that actually be saved to disk 😞.